frontend/src/app/offers/offers.service.ts

```diff
@@ -1,23 +1,28 @@
+import { Offer } from './offers.model';
 import { Injectable } from '@angular/core';
 import { Http, Response } from '@angular/http';
 import { Observable } from 'rxjs/Rx';
 import 'rxjs/add/operator/map';
 
 import { environment } from '../../environments/environment';
-import { Offer } from './offers.model';
+
 
 @Injectable()
 export class OffersService {
   private url = `${environment.apiRoot}/offers/`;
-
   constructor (private http: Http) { }
 
   getOffers() {
     return this.http.get(this.url, { withCredentials: true } )
       .map((res: Response) => res.json());
   }
 
+  getOffer(id: number): Observable<Offer> {
+    return this.http.get(`${this.url}${id}/`, { withCredentials: true } )
+      .map((res: Response) => res.json());
+  }
+
   getDjangoViewUrl(offer: Offer): string {
```

Comment:
How would you name a method that returns a URL for a list of offers?

I'd rename this method to `getDjangoDetailViewUrl` as it returns URL for a specific offer. Then you can have `getDjangoListViewUrl` returning `this.url`.

Reply:
I don't even know if we need this `getDjangoViewUrl` as this component will make it obsolete - maybe we should just delete this view from Django after all.

The other thing is edit page for offer, as this is out of scope of this change and still available only in Django.